middle_end/flambda/simplify/simplify_primitive.ml

```diff
@@ -47,12 +48,12 @@ let try_cse dacc ~original_prim ~simplified_args_with_tys ~min_name_mode
   if not (Name_mode.equal min_name_mode Name_mode.normal) then Not_applied dacc
   else
     let result_var = VB.var result_var in
-    match apply_cse dacc ~original_prim with
+    let args = List.map fst simplified_args_with_tys in
+    match apply_cse dacc ~original_prim:(P.update_args original_prim args) with
```

Comment:
Hmm.  Don't we want to rebind `original_prim` to the version with the simplified arguments?  Otherwise, further down, in the `None` case we'll end up storing the primitive in the environment with the un-simplified arguments.